target_chains/solana/programs/pyth-solana-receiver/tests/program_test/mod.rs

```diff
@@ -40,72 +40,24 @@ pub struct ProgramSimulator {
     /// transaction; otherwise, replayed transactions in different states can return stale
     /// results.
     last_blockhash:        Hash,
-    programdata_id:        Pubkey,
     pub upgrade_authority: Keypair,
     pub genesis_keypair:   Keypair,
 }
 
 impl ProgramSimulator {
     /// Deploys the target chain contract as upgradable
     pub async fn new() -> ProgramSimulator {
-        let mut bpf_data = read_file(
-            std::env::current_dir()
-                .unwrap()
-                .join(Path::new("../../target/deploy/pyth_solana_receiver.so")),
-        );
-
-
-        let mut program_test = ProgramTest::default();
-        let program_key = Pubkey::try_from(ID).unwrap();
-        // This PDA is the actual address in the real world
-        // https://docs.rs/solana-program/1.6.4/solana_program/bpf_loader_upgradeable/index.html
-        let (programdata_key, _) =
-            Pubkey::find_program_address(&[&program_key.to_bytes()], &bpf_loader_upgradeable::id());
+        let mut program_test = ProgramTest::new("pyth_solana_receiver", ID, None);
```

Comment:
i really think you shouldn't change this deployment code. We're doing it this way for a reason (to test upgrades) and you're just going to have to rewrite it again.

Reply:
ok will look into reverting this to original state

Reply:
i reset it back to the original state but just a note that i tried adding back the original update ix and using that test case and it failed with a strange `InvalidAccountData`. I tried debugging the source of this but wasn't able to figure it out. i'll have to look into it later. The test runs fine when i use `ProgramTest::new("pyth_solana_receiver", ID, None);`

Reply:
also is there a repo that tests the program upgrade flow? https://github.com/pyth-network/pyth-client/blob/main/program/rust/src/tests/pyth_simulator.rs also has this setup but it doesn't look like the actual program upgrade is tested there, the upgrade authority is checked for the `upd_permissions` test